pylint/checkers/exceptions.py

```diff
@@ -76,6 +76,26 @@ def _is_raising(body: typing.List) -> bool:
     return False
 
 
+def iterate_ancestors(node: NodeNG) -> typing.Iterator[NodeNG]:
```

Comment:
These two functions can be replaced with `pylint.checkers.utils.find_try_except_wrapper_node`:
```
except_handler = find_try_except_wrapper_node(node)
if isinstance(except_handler, astroid.ExceptHandler):
    # it's found
```

On a quick look that function will need some adjustments to make it work for this use case, but it's better than having another except handler searching routine in the code base.

Reply:
It'll need 2 adjustments. One is adding an argument `stop_on_function: bool = False` and logic for it, that's pretty straightforward. Also consider that it's possible that the fact we haven't had this logic until now causes bugs in other checks, as code that's in a function defined in try-except doesn't have the same behavior as code that's actually in try-except. I won't dwell on that, you can open an issue on that if you'd like.

The second would be for it to find just the `except` clause, and not the `try` clause. Now, I don't know what kind of API you want for that. I can add an argument `only_except: bool = False` but that's not a nice API. Let me know which API you'd like.

Reply:
@PCManticore

I fixed the merge conflict. Besides this issue that I'm waiting on a response on from you, is there anything else that needs to be done before this PR can be merged?

Reply:
Nudge.

Reply:
To answer your question:

* The function should only look in the current scope so it should stop if it encounters a `FunctionDef` or a `ClassDef`.
* We can split this function into two, one for finding `TryExcept` nodes, another for finding the parent `ExceptHandler`, or we can rely on `isinstance` after the function call returned to deduce the same thing.

Reply:
>  We can split this function into two, one for finding `TryExcept` nodes, another for finding the parent `ExceptHandler`.

Okay, I pushed that now.